### PR TITLE
Replace HashValue with HashCode

### PR DESCRIPTION
**Pull request by lptr**
We've introduced the use of Guava's `HashCode` a while ago, then moved to our own `HashCode` implementation for performance reasons, but we never managed to remove `HashValue`. This PR finishes that work.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java [hunk 1]
```diff
@@ -159,11 +158,6 @@ private HashCode getResourceSha1(ExternalResourceName location, boolean revalida
             ExternalResourceReadResult<HashCode> result = resource.withContentIfPresent(inputStream -> {
                 try {
                     String sha = IOUtils.toString(inputStream, StandardCharsets.US_ASCII);
-                    if (sha.length() < 40) {
-                        // servers may return sha-1 with leading 0 stripped, which is not
-                        // supported by HashCode.fromString
-                        return HashCode.fromBytes(HashValue.parse(sha).asByteArray());
-                    }
```
Comment (lptr) [inline subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java:166]: I don't think this is true. Maybe @melix meant Guava's `HashCode.fromString()`?

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QF please.

**Comment by lptr** [comment]
@bot-gradle test RFM.

**Review by britter — APPROVED** [review]
LGTM

**Review by wolfs — APPROVED** [review]
LGTM!

File: subprojects/internal-testing/src/main/groovy/org/gradle/integtests/fixtures/UrlValidator.groovy [hunk 2]
```diff
@@ -76,10 +77,14 @@ class UrlValidator {
      * Asserts that the binary content at the specified url matches the content in the specified File
      */
     static void assertBinaryUrlContent(URL url, File file) {
-        assert compareHashes(url.openStream(), file.newInputStream())
+        url.withInputStream {urlInput ->
+            file.withInputStream { fileInput ->
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/internal-testing/src/main/groovy/org/gradle/integtests/fixtures/UrlValidator.groovy:81]: Yeah, good idea.

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Pull request merged**